include/zephyr/drivers/adc.h

```diff
@@ -707,6 +707,18 @@ __subsystem struct adc_driver_api {
 	uint16_t ref_internal;	/* mV */
 };
 
+/**
+ * @brief Verify that a device has an ADC driver API.
+ *
+ * @param dev a device pointer
+ *
+ * @return true if the device's API is located in the ADC API section.
+ */
+static inline bool adc_device_check(const struct device *dev)
```

Comment:
hey could you make this a normal function so it can be referenced? The implementation could go in the existing `adc_common.c`

Reply:
Sure, what's the added benefit? I wasn't sure this actually remained useful..

Reply:
Shell device filtering: https://github.com/zephyrproject-rtos/zephyr/pull/72037

Reply:
Thinking more about this, I would argue to remove the helper function, as it already is a one-liner. Also, if we move it to `adc_common.c` it isn't possible to link against if `CONFIG_ADC=n` (maybe a non-issue, but still).
The `adc_shell.c` could have a `static` local function just as easy.

```c
static bool adc_device_filter(const struct device *dev)
{
 	return DEVICE_API_IS(adc, dev->api);
}
```

Reply:
yeah good point, let's just drop it then